src/components/WwwFrame/WwwPage.vue

```diff
@@ -64,17 +79,38 @@ export default {
 		};
 	},
 	apollo: {
-		preFetch(config, client, { route }) {
+		preFetch(config, client, { cookieStore, route }) {
+			const getUserData = cookieStore.get(hasLentBeforeCookie) === undefined || cookieStore.get(hasDepositBeforeCookie) === undefined; // eslint-disable-line max-len
+
 			return Promise.all([
 				client.query({ query: hasEverLoggedInQuery }),
 				fetchAllExpSettings(config, client, {
 					query: route?.query,
 					path: route?.path
-				})
+				}),
+				getUserData ? client.query({ query: optimizelyUserDataQuery }) : Promise.resolve()
 			]);
-		},
+		}
 	},
 	created() {
+		let userData = {};
+		const currentHasLentBeforeValue = this.cookieStore.get(hasLentBeforeCookie);
+		const currentHasDepositBeforeValue = this.cookieStore.get(hasDepositBeforeCookie);
+
+		if (!currentHasLentBeforeValue || !currentHasLentBeforeValue) {
+			userData = this.apollo.readQuery({
```

Comment:
Because the `preFetch` doesn't have the same check, I think this `readQuery` will encounter an error when this is called. The first question is: do we need to update the cookie value? If so, then we'll need to make sure the query also runs in the `preFetch` when needed here.

Reply:
Yeah, that is what I am working on right now!